backend/src/main/java/com/example/demo/todo/TodoService.java

```diff
@@ -0,0 +1,58 @@
+package com.example.demo.todo;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@RequiredArgsConstructor
+@Service
+public class TodoService {
+
+    private final TodoRepository todoRepository;
+
+    public List<Todo> getAllTodos() {
+        return todoRepository.findAll();
+    }
+
+    public Todo addTodo(Todo todo) {
+        var tmp = Optional.ofNullable(todoRepository.save(todo));
+        return tmp.orElse(todo);
+    }
+
+    public Optional<Todo> deleteTodo(String id) {
+        return todoRepository.delete(id);
+    }
+
+    public Optional<Todo> nextTodoStatus(Todo todo) {
+        Optional<Todo> todoTmp = todoRepository.findById(todo.getId());
```

Comment:
Das lässt sich auch so schreiben:
```java
return todoRepository.findById(todo.getId())
        .map(todoFromDB -> {
            todoFromDB.setStatus(todoTmp.get().getStatus().toggleStatus());
            return todoFromDB;
        })
        .map(todoRepository::save);
```
Wenn das `setStatus` am `Todo` am Ende `this` zurückgibt, wäre das erste `map` oben auch ein Einzeiler.
```java
public Todo setStatus(Status status) {
    this.status = status;
    return this;
}
```
Das nennt man dann fluent interface.